libraries/AP_Motors/AP_MotorsMatrix.cpp

```diff
@@ -583,42 +583,31 @@ void AP_MotorsMatrix::setup_motors(motor_frame_class frame_class, motor_frame_ty
             }
             break;
 
-        case MOTOR_FRAME_DODECAHEXA: {
-            switch (frame_type) {
-                case MOTOR_FRAME_TYPE_PLUS:
-                    add_motor(AP_MOTORS_MOT_1,     0, AP_MOTORS_MATRIX_YAW_FACTOR_CW,   1);  // TU(Top Up)
-                    add_motor(AP_MOTORS_MOT_2,     0, AP_MOTORS_MATRIX_YAW_FACTOR_CCW,  2);  // TD(Top Down)
-                    add_motor(AP_MOTORS_MOT_3,    60, AP_MOTORS_MATRIX_YAW_FACTOR_CCW,  3);  // URU(Upper Right Up)
-                    add_motor(AP_MOTORS_MOT_4,    60, AP_MOTORS_MATRIX_YAW_FACTOR_CW,   4);  // URD(Upper Right Down)
-                    add_motor(AP_MOTORS_MOT_5,   120, AP_MOTORS_MATRIX_YAW_FACTOR_CW,   5);  // LRU(Lower Right Up)
-                    add_motor(AP_MOTORS_MOT_6,   120, AP_MOTORS_MATRIX_YAW_FACTOR_CCW,  6);  // LRD(Lower Right Down)
-                    add_motor(AP_MOTORS_MOT_7,   180, AP_MOTORS_MATRIX_YAW_FACTOR_CCW,  7);  // BU(Bottom Up)
-                    add_motor(AP_MOTORS_MOT_8,   180, AP_MOTORS_MATRIX_YAW_FACTOR_CW,   8);  // BD(Bottom Down)
-                    add_motor(AP_MOTORS_MOT_9,  -120, AP_MOTORS_MATRIX_YAW_FACTOR_CW,   9);  // LLU(Lower Left Up)
-                    add_motor(AP_MOTORS_MOT_10, -120, AP_MOTORS_MATRIX_YAW_FACTOR_CCW, 10);  // LLD(Lowoe Left Down)
-                    add_motor(AP_MOTORS_MOT_11,  -60, AP_MOTORS_MATRIX_YAW_FACTOR_CCW, 11);  // ULU(Upper Left Up)
-                    add_motor(AP_MOTORS_MOT_12,  -60, AP_MOTORS_MATRIX_YAW_FACTOR_CW,  12);  // ULD(Upper Left Down)
+        case MOTOR_FRAME_DODECAHEXA:
```

Comment:
I don't think we can just use a Y cable between the upper and lower motors on the same arm because then we would end up with no yaw control.  I think every motor will need to be independently controlled.
I've done some fix ups in this branch.  I think it will fly but there seems to be a problem with the reporting of motors 9 ~ 12 to the ground station.: https://github.com/rmackay9/rmackay9-ardupilot/commits/muramura-dodeca